### ISSUE TITLE
Make it more verbose if prediction_activation and criter_spec are not compatible

### ISSUE THREAD

**Issue by rfratila** [Aifred-Health/Vulcan#81]
pred_activation = nn.Softmax and criter_spec = nn.CrossEntropyLoss should not be used together since softmax is already implemented in the loss.

Valid configurations
pred_activation = None and criter_spec = nn.CrossEntropyLoss()
pred_activation = nn.LogSoftmax() and writer_spec = nn.NLLLoss()

**Comment by nirtiac** [comment]
addressed in #130

**Comment by nirtiac** [comment]
finally addressed in #163
